Mathlib/Topology/ProperMap.lean

```diff
@@ -136,6 +137,60 @@ lemma IsProperMap.ultrafilter_le_nhds_of_tendsto (h : IsProperMap f) ⦃𝒰 : U
     (hy : Tendsto f 𝒰 (𝓝 y)) : ∃ x, f x = y ∧ 𝒰 ≤ 𝓝 x :=
   (isProperMap_iff_ultrafilter.mp h).2 hy
 
+/-- The composition of two proper maps is proper. -/
+lemma IsProperMap.comp (hf : IsProperMap f) (hg : IsProperMap g) :
+    IsProperMap (g ∘ f) := by
+  refine ⟨by continuity, fun ℱ z h ↦ ?_⟩
+  rw [MapClusterPt, ← map_map] at h
+  rcases hg.clusterPt_of_mapClusterPt h with ⟨y, hy1, hy2⟩
+  rcases hf.clusterPt_of_mapClusterPt hy2 with ⟨x, hx1, hx2⟩
+  use x
+  rw [← hy1, ← hx1]
+  exact ⟨rfl, by assumption⟩
+
+/-- If the composition of two continuous functions `g ∘ f` is proper and `f` is surjective,
+then `g` is proper. -/
+lemma isProperMap_of_comp_of_surj (hf : Continuous f)
+    (hg : Continuous g) (hgf : IsProperMap (g ∘ f)) (f_surj : f.Surjective) : IsProperMap g := by
+  refine ⟨hg, fun ℱ z h ↦ ?_⟩
+  rcases hgf with ⟨_, h'⟩
+  rw [← inf_top_eq ℱ, ← principal_univ, ← f_surj.range_eq, ← ℱ.map_comap f, MapClusterPt,
+    map_map, ← MapClusterPt] at h
+  rcases h' h with ⟨x, hx1, hx2⟩
+  refine ⟨f x, hx1, neBot_of_comap (m := f) (NeBot.mono hx2 ?_)⟩
+  calc
+    𝓝 x ⊓ comap f ℱ
+      ≤ comap f (𝓝 (f x)) ⊓ comap f ℱ := inf_le_inf_right _ (hf.tendsto x).le_comap
+    _ = comap f (𝓝 (f x) ⊓ ℱ) := comap_inf.symm
+
+/-- If the composition of two continuous functions `g ∘ f` is proper and `g` is injective,
+then `f` is proper. -/
+lemma isProperMap_of_comp_of_inj {f : X → Y} {g : Y → Z} (hf : Continuous f) (hg : Continuous g)
+    (hgf : IsProperMap (g ∘ f)) (g_inj : g.Injective) : IsProperMap f := by
+  refine ⟨hf, fun ℱ y h ↦ ?_⟩
+  rcases hgf with ⟨_, h'⟩
+  have : MapClusterPt (g y) ℱ (g ∘ f) := by
+    refine h.map (hg.tendsto y) ?_
+    rw [← map_map, Tendsto]
```

Comment:
```suggestion
  have : MapClusterPt (g y) ℱ (g ∘ f) := h.map hg.continuousAt tendsto_map
```

Two things here:
- As you noticed, `ContinuousAt` unfolds to a `Tendsto` statement. Still, it's a bit cleaner to use `hg.continuousAt` directly, rather than `hg.tendsto`.
- It turns out the functoriality of `map` is definitional, so you can (ab)use it and use [Filter.tendsto_map](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Filter/Basic.html#Filter.tendsto_map) directly. If you're not happy with this feel free to ignore.